picking_good_slis.md

```diff
@@ -0,0 +1,92 @@
+# Picking Good SLIs
+
+## SRE's Philosophy on Monitoring
+
+A core goal of SRE is to ensure the availability of a service. "Availability"
+here means much more than simply "is my application running." Effectively
+monitoring a modern cloud application requires insight into the application's
+response time, error rate, traffic level, and more.
+
+Furthermore, the scope of monitoring should not be limited solely for use in
+deciding when to page an operations team. Metrics should be closely tied to
+business objectives, informing decisions on what features to prioritize and
+when to focus on adding new features versus improving reliability.
+
+To make this vision a reality, SRE teams rely on two key concepts:
+  * Service Level Indicators (SLIs) - A quantifiable metric that measures a
+    specific aspect of a service's availability.
+  * Service Level Objectives (SLOs) - A measurable, specific goal for a
+    given service's availability over time, based on one or more SLIs.
+
+As a starting point on towards realizing this goal, this document aims to
+provide guidance on selecting the right SLIs for your service.
+
+## Selecting SLIs
+
+For SRE teams looking to start monitoring their services, start by identifying
+SLIs to measure. Focus on measuring the basics to start, can refine them over
+time as you gain more experience operating the service.
+
+A good starting point is to attempt to implement an SLI for each of the four
+"Golden Signals".
+
+## The Four Golden Signals of Monitoring
```

Comment:
This section is a good summary of Google's SRE book. I'd like to see some advice on the mechanics of instrumentation to guide teams.

One of the things that comes to mind immediately would be a caution against "long inferences." A "long inference" can be explained by analogy: Measuring a laptop's CPU temperature by using the CD-ROM thermal sensor will not yield a CPU temperature measurement that has high fidelity.

This document is the perfect place to capture these kinds of pitfalls for the various kinds of golden signals.

Some other pitfalls that could be explored:

* Where to start and stop a stopwatch for latency measurements (when is the request "over"?)
* Long long are messages in a message queue before being processed vs message queue ingestion rate

Finally, I'd like to see a caution against hooking up metrics to alerts for the sake of doing _something_ at the start of an SRE team's journey. This goes to the "top-down thinking" comment I made elsewhere: if SLOs drive SLIs, then it is less likely that alerts are attached to things that "don't matter."